src/OpenCV/ImgProc/GeometricImgTransform.hsc

```diff
@@ -297,3 +298,37 @@ getRotationMatrix2D center angle scale = unsafeCoerceMat $ unsafePerformIO $
   where
     c'angle = realToFrac angle
     c'scale = realToFrac scale
+
+{- | Applies a generic geometrical transformation to an image.
+
+<http://docs.opencv.org/3.0-last-rst/modules/imgproc/doc/geometric_transformations.html#remap>
+-}
+remap
+    :: Mat ('S [inputHeight, inputWidth]) inputChannels inputDepth
+    -> Mat ('S [outputHeight, outputWidth]) ('S 2) ('S Float)
+    -> InterpolationMethod
+    -> BorderMode
+    -> Mat ('S [outputHeight, outputWidth]) inputChannels inputDepth
+remap src mapping interpolationMethod borderMode =
+  unsafeCoerceMat $ unsafePerformIO $ do
+    dst <- newEmptyMat
+    withPtr src $ \srcPtr ->
+      withPtr dst $ \dstPtr ->
+        withPtr mapping $ \mappingPtr ->
+          withPtr borderValue $ \borderValuePtr ->
+            [CU.block| void {
```

Comment:
In general we do not want to make unsafe foreign calls (`CU.block`), unless the foreign call is very fast. While `cv::remap` is highly optimized it isn't super fast. It also highly depends on the size of the input matrices.

Unsafe calls to the FFI block the OS capacility associated with the FFI thread until the call completes.

This is not really relevant if you change the quasiquoter from `CU.block` to `cvExcept`.